text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
@mahkoh 

> If you return an iterator, then, on linux, there is no reason to do any allocations.

Right. That's exactly the point that @XMPPwocky already raised. The RFC proposes moving to an iterator, and I agree with @XMPPwocky that further looking into `Cow` here is worth doing to avoid even more allocations.